aws/resource_aws_dx_gateway_association_test.go

```diff
@@ -182,38 +182,6 @@ func testSweepDirectConnectGatewayAssociations(region string) error {
 	return nil
 }
 
-func TestAccAwsDxGatewayAssociation_deprecatedSingleAccount(t *testing.T) {
-	resourceName := "aws_dx_gateway_association.test"
-	resourceNameDxGw := "aws_dx_gateway.test"
-	resourceNameVgw := "aws_vpn_gateway.test"
-	rName := fmt.Sprintf("terraform-testacc-dxgwassoc-%d", acctest.RandInt())
-	rBgpAsn := randIntRange(64512, 65534)
-
-	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheck(t) },
-		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckAwsDxGatewayAssociationDestroy,
-		Steps: []resource.TestStep{
-			{
-				Config: testAccDxGatewayAssociationConfig_deprecatedSingleAccount(rName, rBgpAsn),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckAwsDxGatewayAssociationExists(resourceName),
-					resource.TestCheckResourceAttrPair(resourceName, "dx_gateway_id", resourceNameDxGw, "id"),
-					resource.TestCheckResourceAttrPair(resourceName, "vpn_gateway_id", resourceNameVgw, "id"),
-					resource.TestCheckResourceAttrSet(resourceName, "dx_gateway_association_id"),
-					resource.TestCheckNoResourceAttr(resourceName, "associated_gateway_id"),
-					resource.TestCheckResourceAttr(resourceName, "associated_gateway_type", "virtualPrivateGateway"),
-					testAccCheckResourceAttrAccountID(resourceName, "associated_gateway_owner_account_id"),
-					testAccCheckResourceAttrAccountID(resourceName, "dx_gateway_owner_account_id"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_prefixes.#", "1"),
-					tfawsresource.TestCheckTypeSetElemAttr(resourceName, "allowed_prefixes.*", "10.255.255.0/28"),
-					testAccCheckAwsDxGatewayAssociationStateUpgradeV0(resourceName),
```

Comment:
This is the only call to `testAccCheckAwsDxGatewayAssociationStateUpgradeV0` so it needs to go.

Reply:
This was actually a good catch by the linter. We still want to acceptance test the state upgrade function since it requires an API call. Reinstated this test using a 3.0 valid configuration using `associated_gateway_id` and updated `testAccCheckAwsDxGatewayAssociationStateUpgradeV0()` to appropriately setup the state upgrade using the new attribute.

Output from acceptance testing:

```
--- PASS: TestAccAwsDxGatewayAssociation_V0StateUpgrade (2605.48s)
```